### PR TITLE
Refactor check env and fossa cli

### PR DESCRIPTION
**Pull request by elaguerta-nr**
# Refactor check-env and FOSSA CLI
Refactor all workflows to pull out duplicate code from `check_env` and `fossa` jobs. Fix bugs that suppressed errors from FOSSA analyze, and incorrectly set the `fossaAnalyzeResult` custom property. 

### Refactors
- Pull out FOSSA CLI duplicated code into a [composite action](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) at [.github/composite/fossa-composite/action.yml](https://github.com/newrelic/.github/blob/a95054112259edb031c2147731bd054fc97e36e5/.github/composite/fossa-composite/action.yml). 
  - A composite action enables the FOSSA steps to be called in the same job as the calling job. This allows the FOSSA CLI to to run after required configuration steps such as installing package managers. 
- Pull out `check-env` duplicated code into a [reusable workflow](https://docs.github.com/en/actions/using-workflows/reusing-workflows) at [.github/workflows/fossa-check-env.yml](https://github.com/newrelic/.github/blob/a95054112259edb031c2147731bd054fc97e36e5/.github/workflows/fossa-check-env.yml)
    - A reusable workflow runs as a separate job. This allows it to be tracked and monitored separately from the FOSSA job. It also minimizes passing of required secrets from the caller to the `check-env` job. 

### Bugfixes
- Change error checking of fossa CLI commands to [search for `[ERROR]` in error logs](https://github.com/newrelic/.github/pull/73/files#diff-0072ff56e6b09e5400af2314f02e7614c23f49543891219ca25e8b183a901495R40) (exact match)
  - [previous check](https://github.com/newrelic/.github/pull/73/files#diff-f3fcd3e07be15dff6bf1f07b3d81cc2d4c809ebd35581788e5b05e27b33fa5eaL71) was incorrectly matching on an [informational log with the word "error" ](https://github.com/fossas/fossa-cli/blob/c43e2e362c03154d0d0cdb30e5fa80c1fd8bdda2/src/App/Fossa/Analyze/ScanSummary.hs#L166) 
  - We are resorting to text matching because we couldn't find an exit code for `fossa analyze` that indicates an execution error. Failing error codes indicate a policy violation, which we don't want to block on at this time.
- [Initialize `HAS_FOSSA_TARGETS` and `FOSSA_ANALYZE_RESULT` env variables to ""](https://github.com/newrelic/.github/pull/73/files#diff-a35a493af906a58011d7c0de44a2150c8b864af2e4a43eec58c4bdbf26ea0f3fR21-R22)
  - This ensures that the values of custom properties get set to default values if the FOSSA job encounters errors.
- fossa-elixir.yml
    - [pass`elixir-version` input variable](https://github.com/newrelic/.github/pull/73/files#diff-cd334ada13e0465707133e3e98eecea5231eb90edfffec5bb79c2d53dbc3f49bL32)
    - fix "ANALZYE" [typo](https://github.com/newrelic/.github/pull/73/files#diff-cd334ada13e0465707133e3e98eecea5231eb90edfffec5bb79c2d53dbc3f49bL71-L72) that suppressed fossa analyze errors (by replacing all FOSSA work with call to shared composite workflow)
- fossa-ruby.yml
   - fix the same ANALZYE [typo](https://github.com/newrelic/.github/pull/73/files#diff-f3fcd3e07be15dff6bf1f07b3d81cc2d4c809ebd35581788e5b05e27b33fa5eaL69-L70)
- fossa-scala.yml
  - fix the same ANALZYE [typo](https://github.com/newrelic/.github/pull/73/files#diff-68e0e55a6799914c803c39ad998789531ab2a1d4f854613185c91d6fa22ae66fL71-L72)

### CODE REVIEW

**Review by lucasgonze — APPROVED** [review]
No issues jump out at me.

File: .github/composite/fossa-composite/action.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            echo "HAS_FOSSA_TARGETS=True" >> "$GITHUB_ENV"
+        else
+            echo "::warning::Fossa did not find any analysis targets."
+            echo "HAS_FOSSA_TARGETS=False" >> "$GITHUB_ENV"
+            echo "FOSSA_ANALYZE_RESULT=N/A" >> "$GITHUB_ENV"
+        fi
+
+    - id: fossa-analyze
+      name: Run fossa analyze
+      shell: bash
```
Comment (cmboling) [inline .github/composite/fossa-composite/action.yml:34]:
```suggestion

```
small nit - you can remove these instances and call it under line 4 for the entire workflow

Comment (elaguerta-nr) [reply .github/composite/fossa-composite/action.yml:34]: great call! I moved the shell definition up for the entire workflow and tested the new syntax [here](https://github.com/newrelic-csec/docs-website/actions/runs/9290297801) on a test repo. thanks for catching this.

Comment (cmboling) [reply .github/composite/fossa-composite/action.yml:34]: Okie dokie this looks great. Overall, this is well done! 🚀 🚀 🚀 🚀 🚀

**Review by cmboling — APPROVED** [review]

**Pull request merged**